app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,125 @@
+# TASKS = [
+#     {name: "exercise (begrudgingly)"},
+#     {name: "grocery shop"},
+#     {name: "tend to Kirby, the small dog"},
+#     {name: "force brain to learn"},
+#     {name: "fight off sense of future dread"},
+#     {name: "make dinner"},
+#     {name: "pick up packages"}
+# ]
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def update
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      head :not_found
+      return
+    elsif @task.update(
+        name: params[:task][:name],
+        description: params[:task][:description],
+        completed_at: params[:task][:completed_at]
+    )
+      redirect_to task_path # go to the index so we can see the book in the list
+      return
+    else # save failed :(
+    render :edit # show the new book form view again
```

Comment:
It's helpful to include a HTTP status indicating failure.

```suggestion
    render :edit, status: :bad_request # show the new book form view again
```